samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java

```diff
@@ -27,82 +30,105 @@
+					Logger.info("NotificationService:processMessage: hashValue is not matching - " + requestHash);
 				} else {
-					Map<String, Object> configMap = (Map<String, Object>) tmp.get(Constant.CONFIG);
-					topic = (String) configMap.getOrDefault(Constant.TOPIC, "");
+					Map<String, Object> tmp = (Map<String, Object>) requestMap.get(Constant.NOTIFICATION);
+					if (tmp.get(Constant.IDS) != null) {
+						deviceIds = (List<String>) tmp.get(Constant.IDS);
+					}
+					if(tmp.get(Constant.MODE).equals("phone")) {
+						if(deviceIds != null) {
+							Map<String, Object> templateMap = (Map<String, Object>) tmp.get(Constant.TEMPLATE);
+							String smsData = (String) templateMap.get(Constant.DATA);
+							isSuccess = sendSmsNotification(deviceIds, smsData);
+						} else {
+							Logger.info("mobile numbers not provided");
+						}
+					} else {
+						if (deviceIds == null) {
+							Map<String, Object> configMap = (Map<String, Object>) tmp.get(Constant.CONFIG);
+							topic = (String) configMap.getOrDefault(Constant.TOPIC, "");
+						}
+						notificationMap.put(Constant.RAW_DATA, mapper.writeValueAsString(tmp.get(Constant.RAW_DATA)));
+						Logger.info("NotificationService:processMessage: calling send notification ");
+						isSuccess = notifyDevice(notificationMap, deviceIds, topic);
+					}
+					if (isSuccess) {
+						Logger.info("Notification sent to device successfully.");
+					} else {
+						Logger.info("Notification sent failure for device or token " + deviceIds + " _" + topic);
+					}
 				}
-				notificationMap.put(Constant.RAW_DATA, mapper.writeValueAsString(tmp.get(Constant.RAW_DATA)));
 			} else {
 				Logger.info("NotificationService:processMessage procide actioname is incorrect: " + actionValue);
 			}
 		} else {
 			Logger.info("NotificationService:processMessage event data map is either null or empty");
 		}
-		Map<String, Object> objectMap = (Map<String, Object>) message.get(Constant.OBJECT);
-		if (!requestHash.equals((String) objectMap.get(Constant.ID))) {
-			Logger.info("NotificationService:processMessage: hashValue is not matching - " + requestHash);
-		} else {
-			Logger.info("NotificationService:processMessage: calling send notification ");
-			boolean isSuccess = notify(notificationMap, deviceIds, topic);
-			if (isSuccess) {
-            Logger.info("Notification sent to device successfully.");
-			} else {
-			Logger.info("Notification sent failure for device or token " + deviceIds + " _" + topic); 	
-			}
-		}
+	}
 
+	private boolean sendSmsNotification(List<String> deviceIds, String smsText) {
+		return smsProvider.bulkSms(deviceIds, smsText);
 	}
 
-	private boolean notify(Map<String, String> notificationMap, List<String> deviceIds, String topic) {
+	private boolean notifyDevice(Map<String, String> notificationMap, List<String> deviceIds, String topic) {
 		if (deviceIds != null && deviceIds.size() > 0) {
 			if (deviceIds.size() <= 100) {
```

Comment:
if size 100 is decided, it must just chunkified and a simple for loop to batch. The current one has if-else - consider rewrite.